src/herder/HerderImpl.cpp

```diff
@@ -130,6 +130,15 @@ HerderImpl::trackingConsensusCloseTime() const
 void
 HerderImpl::setState(State st)
 {
+    bool initState =
+        st == HERDER_BOOTING_STATE || st == HERDER_TRACKING_LCL_STATE;
```

Comment:
There might be a problem when the app runs "twice" in the `!MODE_AUTO_STARTS_OVERLAY` mode:
```
runApp(Application::pointer app):
if (!app->getConfig().MODE_AUTO_STARTS_OVERLAY)
        {
            app->getHerder().restoreState();


HerderImpl::restoreState() -> HerderImpl::restoreSCPState() -> setTrackingSCPState(lcl.header.ledgerSeq, lcl.header.scpValue, false);

```

How can we guarantee that the app didn't move to `HERDER_TRACKING_NETWORK_STATE` or `HERDER_SYNCING_STATE`  when `ApplicationImpl::start()` invoked `mHerder->restoreState();` when the app gets started the first time?